src/stream/src/executor/changelog.rs

```diff
@@ -40,27 +47,47 @@ impl Execute for ChangeLogExecutor {
     }
 }
 impl ChangeLogExecutor {
-    pub fn new(ctx: ActorContextRef, input: Executor, need_op: bool) -> Self {
+    pub fn new(
+        ctx: ActorContextRef,
+        input: Executor,
+        need_op: bool,
+        all_vnode_count: usize,
+        vnodes: Bitmap,
+    ) -> Self {
+        let changelog_row_id_generator = ChangelogRowIdGenerator::new(vnodes);
+        let pk_indices = input.pk_indices().to_vec();
         Self {
-            _ctx: ctx,
+            ctx,
             input,
             need_op,
+            all_vnode_count,
+            pk_indices,
+            changelog_row_id_generator,
         }
     }
 
     #[try_stream(ok = Message, error = StreamExecutorError)]
-    async fn execute_inner(self) {
+    async fn execute_inner(mut self) {
         let input = self.input.execute();
         #[for_await]
         for msg in input {
             let msg = msg?;
             match msg {
                 Message::Chunk(chunk) => {
+                    let data_chunk = chunk.data_chunk();
+                    let vnodes = VirtualNode::compute_chunk(
+                        data_chunk,
+                        &self.pk_indices,
```

Comment:
This is very wrong. Vnode should be calculated based on the distribution key instead of stream key.